### ISSUE TITLE
I came back again...orn. FriedLiver stuck at SIFT.

### ISSUE THREAD

**Issue by Jie2World** [nonlinear1/BundleFusion_Ubuntu_V0#6]
I make this project work, but face the same awkward  problem like win10. FriedLiver stuck when it load sens file and begin to compute SIFT with my poor GPU. The error is

FriedLiver/Source/SiftGPU/SiftPyramid.cpp(387):cudaSafeCall() Run time API error 715: an illegal instruction was encountered.

Any idea to solve it, or just compute without GPU, or close multi-threaded during runing? 
Hope help me.

**Comment by RigerLee** [comment]
I have the same issue on TITAN RTX, but it works fine on 1050Ti and 1080Ti. So maybe you should try GTX series.

**Comment by Quitino** [comment]
I have tried on GTX1050,An embarrassing problem has occurred and there is not enough memory.
![2019-11-29 08-49-55屏幕截图](https://user-images.githubusercontent.com/38552528/69836656-3fe2cd00-1285-11ea-848a-9ed224124f4c.png)

**Comment by southpaw-wu** [comment]
The same problem will be encountered with 1080Ti, but after a long run time

**Comment by RigerLee** [comment]
> The same problem will be encountered with 1080Ti, but after a long run time

Yes, sometimes you need to modify config and .sens to make it fit your GPU memory.

**Comment by CAJvon** [comment]
Hi,
I encountered the same problem, the program dumped and it showed "Segmentation fault (core dumped)", and my gpu is also TITAN. I think it stuck at SIFT, too.
But in my opinion, the gpu doesn't matter, because I use the same config.txt and data.sens on Win10, it works.
So I think there will be a better solution than replace the GPU. Maybe it's about CUDA?
Hope someone helps.

**Comment by CAJvon** [comment]
Hi, friends:
I encountered the same problem, the program dumped and it showed `Segmentation fault (core dumped)`, and my gpu is also TITAN. I think it stuck at SIFT, too.
But in my opinion, the gpu doesn't matter, because I use the same config.txt and data.sens on Win10, it works.
So I think there will be a better solution than replace the GPU. Maybe it's about CUDA?
Hope someone helps.

**Comment by liangyongshi** [comment]
" xxxx/BundleFusion_Ubuntu_V0/FriedLiver/Source/SiftGPU/SiftPyramid.cpp(387) : cudaSafeCall() Runtime API error 715: an illegal instruction was encountered."

I also encountered this problem， any one solved?

ubuntu20.04,  cuda11.6,  opencv4.5.0(with cuda), pangolin 0.6, RTX3090

**Comment by liangyongshi** [comment]
感觉原因是显卡和cuda版本比较高的问题，代码里面是要选算力的，rtx30系列的sm是86，而老的版本里面是没有这个选项的，我还在debug，找到几个问题，目前到这一步了，一直搞不定。

FriedLiver/Source/DepthSensing/CameraUtil.cu(1254): error: calling a constexpr __host__ function("min") from a __global__ function("convertEdgeMaskToFloatDevice") is not allowed. The experimental flag '--expt-relaxed-constexpr' can be used to allow this.

**Comment by nasireddypavan** [comment]
Hi ,

i'm Facing the same issue.

**Comment by liangyongshi** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，已经收到您的邮件，无法亲自回复，还请见谅。
